src/main/NWNX/API/Events/PartyEvents.cs

```diff
@@ -0,0 +1,329 @@
+using NWN.API;
+using NWN.Core;
+using NWN.Core.NWNX;
+
+namespace NWNX.API.Events
+{
+  public class PartyEvents
+  {
+    [NWNXEvent("NWNX_ON_PARTY_LEAVE_BEFORE")]
+    public class OnPartyLeaveBefore : EventSkippable<OnPartyLeaveBefore>
```

Comment:
Since this is already in `PartyEvents`, these class names can be simplified.

```suggestion
    public class OnLeaveBefore : EventSkippable<OnPartyLeaveBefore>
```